kerby-kerb/kerb-client/src/main/java/org/apache/kerby/kerberos/kerb/client/KrbClientBase.java

```diff
@@ -271,15 +271,22 @@ public void storeTicket(TgtTicket tgtTicket,
      */
     public void storeTicket(SgtTicket sgtTicket, File ccacheFile) throws KrbException {
         LOG.info("Storing the sgt to the credential cache file.");
-        if (!ccacheFile.exists()) {
+        boolean isFreshNew = !ccacheFile.exists() || (ccacheFile.length() == 0);
+        if (isFreshNew) {
             createCacheFile(ccacheFile);
         }
         if (ccacheFile.exists() && ccacheFile.canWrite()) {
-            CredentialCache cCache = new CredentialCache();
+            
             try {
-                cCache.load(ccacheFile);
-                cCache.addCredential(new Credential(sgtTicket, sgtTicket.getClientPrincipal()));
-                cCache.setPrimaryPrincipal(sgtTicket.getClientPrincipal());
```

Comment:
Why is this line removed?

Reply:
When ccache exists and is correctly loaded, the cCache primaryPrincipal field is already populated. I think that adding an SGT should not change the cCache primary principal. I think that, setting the primary principal, is onely needed whena  fresh new cache file is created.